sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -72,137 +86,129 @@ export class CASSupplierIntegrationService {
     for (const casSupplier of casSuppliers) {
       const summary = new ProcessSummary();
       parentProcessSummary.children(summary);
-      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      summary.info(`Processing student CAS supplier ID: ${casSupplier.id}.`);
       try {
-        const supplierResponse = await this.casService.getSupplierInfoFromCAS(
-          auth.access_token,
-          casSupplier.student.sinValidation.sin,
-          casSupplier.student.user.lastName,
+        // Check the current status of the student data and its supplier information.
+        const evaluationResult = await this.evaluateCASSupplier(
+          casSupplier,
+          auth,
+        );
+        summary.info(
+          `CAS evaluation result status: ${evaluationResult.status}.`,
+        );
+        // Decide the process to be executed.
+        const processor = this.getCASSupplierProcess(evaluationResult.status);
+        // Execute the process.
+        const processResult = await processor.process(
+          casSupplier,
+          evaluationResult,
+          auth,
+          summary,
         );
-        if (await this.updateSupplier(supplierResponse, summary, casSupplier)) {
+        if (processResult.isSupplierUpdated) {
           suppliersUpdated++;
         }
       } catch (error: unknown) {
-        summary.error(
-          "Error while requesting or updating CAS suppliers.",
-          error,
-        );
+        // Log the error and allow the process to continue checking the
+        // remaining student suppliers.
+        summary.error("Unexpected error while processing supplier.", error);
       }
     }
     return suppliersUpdated;
   }
 
   /**
-   * Updates supplier if finds an item from the response with an active address.
-   * @param supplierResponse CAS supplier response.
-   * @param summary log summary.
-   * @param casSuppliers pending CAS suppliers.
-   * @returns true if updated a record.
+   * Get the processor associated to the CAS evaluation status result.
+   * @param evaluationResult evaluation result status.
```

Comment:
I believe the `param` needs an update here.

Reply:
Updated.